src/Stats.php

```diff
@@ -6,137 +6,46 @@
-     *                     the duration will be converted to a human-readable format. If the sync start or end time is empty,
-     *                     returns false.
-     */
-    public function get_sync_duration(bool $human_time = false): float|string|false {
-        if (
-            empty($this->sync_start)
-            || empty($this->sync_end)
-        ) {
-            return false;
-        }
-
-        $duration = round((float)$this->sync_end->format('U.u') - (float)$this->sync_start->format('U.u'), 4);
-        if ($human_time) {
-            return $this->human_time_diff_microseconds(0, $duration);
-        }
-
-        return $duration;
-    }
+    use DB;
 
     /**
-     * Add a new action with its details.
+     * The group name for the sync process.
      *
-     * @param int $id
-     * @throws \Exception
+     * @var string
      */
-    public function add_action(int $id): void
-    {
-        $this->actions[$id] = [
-            'id'    => $id,
-            'start' => current_datetime(),
-        ];
-        $this->save();
-    }
+    private string $group_name;
 
     /**
-     * End an action and sets endtime as well as duration.
+     * Stats constructor.
      *
-     * @param int $id
-     * @return void
-     * @throws \Exception
+     * @param string $group_name The group name for the sync process.
      */
-    public function end_action(int $id): void
-    {
-        if (!isset($this->actions[$id])) {
-            return;
-        }
-
-        $this->actions[$id]['status'] = 'success';
-        $this->actions[$id]['end'] = current_datetime();
-        $this->save();
+    public function __construct(string $group_name) {
+        $this->group_name = $group_name;
     }
 
     /**
-     * Marks an action as failed.
+     * Gets the sync duration.
      *
-     * @param int $id The ID of the action to mark as failed.
-     * @param string $error_message The error message associated with the failed action.
-     * @return void
-     * @throws \Exception If the action ID is not found in the actions array.
+     * @param bool $human_time Whether to return human-readable time.
+     * @return float|string|false Duration or false if not available.
      */
-    public function mark_action_as_failed(int $id, string $error_message): void
-    {
-        if (!isset($this->actions[$id])) {
-            throw new \Exception("Action ID $id not found");
-        }
+    public function get_sync_duration(bool $human_time = false): float|string|false {
+        $query = $this->db()->prepare(
+            "SELECT MIN(start) as sync_start, MAX(end) as sync_end 
+            FROM {$this->get_chunks_table_name()} 
+            WHERE `group` = %s",
+            $this->group_name
+        );
 
-        $this->actions[$id]['status'] = 'failed';
-        $this->actions[$id]['error_message'] = $error_message;
-        $this->save();
```

Comment:
you have added a function inside the chunk class why not using it?

Reply:
Because this is the sync duration for the whole group. With MIN(start) and MAX(end) I get the values I need to calculate the whole sync process. The duration method in the Chunk Entity just calculate the duration for a single chunk.